enterprise/frontend/src/embedding-sdk/cli/steps/create-models-and-xrays.ts

```diff
@@ -0,0 +1,58 @@
+import ora from "ora";
+
+import { createXrayDashboardFromModel } from "embedding-sdk/cli/utils/xray-models";
+import type { DashboardId } from "metabase-types/api";
+
+import type { CliStepMethod } from "../types/cli";
+import { createModelFromTable } from "../utils/create-model-from-table";
+
+export const createModelsAndXrays: CliStepMethod = async state => {
+  const { instanceUrl = "", databaseId, cookie = "", tables = [] } = state;
+
+  if (databaseId === undefined) {
+    return [{ type: "error", message: "No database selected." }, state];
+  }
+
+  const spinner = ora("Creating models…").start();
+
+  try {
+    // Create a model for each table
+    const modelIds = await Promise.all(
+      tables.map(table =>
+        createModelFromTable({
+          table,
+          databaseId,
+          cookie,
+          instanceUrl,
+        }),
+      ),
+    );
+
+    spinner.start("X-raying your data to create dashboards...");
+
+    const dashboardIds: DashboardId[] = [];
+
+    // We create dashboard sequentially to prevent multiple
+    // "Automatically Generated Dashboards" collection from being created.
```

Comment:
This is because of an interesting edge-case in x-rays: if you x-ray 5 dashboards at the same time, it actually creates 5 different collections called "Automatically Generated Dashboards" - I guess it's because the collection with that name/flag hasn't been created yet. Creating it one-by-one fixes this issue.